server/paymentProviders/opencollective/collective.js

```diff
@@ -63,32 +43,14 @@ paymentMethodProvider.processOrder = async order => {
   const fromCollectiveHost = await order.fromCollective.getHostCollective();
   const collectiveHost = await order.collective.getHostCollective();
 
-  // If the paymentMethod is the one of the host to make a payment on behalf of another of its collective,
-  // we need to use the payment method of the fromCollective (to make sure it has enough funds)
-  if (
-    fromCollectiveHost &&
-    order.paymentMethod.CollectiveId === fromCollectiveHost.id &&
-    fromCollectiveHost.id === collectiveHost.id
-  ) {
```

Comment:
Deleting these conditions are they're not making sense to me and preventing to check the balance.